### DISCUSSION TITLE
Access Denied when I use "security"="object == user"

### DISCUSSION THREAD

**Discussion by kevinG73** [symfony/symfony#42436] [Q&A] [answered]
Hello,
I use API platform and symfony 5.3, I built a JSON Authentication Endpoint using this method https://symfony.com/doc/current/security/json_login_setup.html and  it works without problem except when I added    `"security"="object == user"` :

```php
class User implements UserInterface, PasswordAuthenticatedUserInterface
```

```php
 *     itemOperations = {
 *         "put"={
 *              "security"="object == user"
 *          }
 *     },
 *     normalizationContext = {
 *          "groups" = {"User:read"}
 *     },
 *     denormalizationContext = {
 *          "groups"= {"User:write"}
 *     }
 * )
 ```
 I get this error : 
 ```
 {
  "type": "https:\/\/tools.ietf.org\/html\/rfc2616#section-10",
  "title": "An error occurred",
  "detail": "Access Denied.",
}
```
But I don't know why because my user is logged currently, someone could help me please ?

**Comment by NielsJanssen** [comment, marked as answer]
A possible cause is that `object` and `user` are not actually the same objects, but still contain mostly the same data. So comparing the objects directly wouldn't work. Assuming your user objects implement a unique username you could try `object.username === user.username` as a comparison.

If that doesn't work, try using the profiler to debug your request. The security tab contains a lot of useful information which of the security checks actually denies access.

**Reply by kevinG73** [reply]
Thank you , it works
```php
 "security"="object.getUsername() === user.getUsername()"
```
